### PR TITLE
pass macro dep to cppcheck

### PR DESCRIPTION
**Pull request by claireyywang**
part of the effort to pin cppcheck 2.0, ros2/ros2#942
Signed-off-by: claireyywang <22240514+claireyywang@users.noreply.github.com>

### CODE REVIEW

**Comment by claireyywang** [comment]
CI 
linux [Build Status](https://ci.ros2.org/job/ci_linux/11329/)
linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/6567/)
osx [Build Status](https://ci.ros2.org/job/ci_osx/9293/)
windows [Build Status](https://ci.ros2.org/job/ci_windows/11225/)

**Comment by claireyywang** [comment]
CI again 
linux [Build Status](https://ci.ros2.org/job/ci_linux/11332/)
linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/6570/)
osx [Build Status](https://ci.ros2.org/job/ci_osx/9296/)
windows [Build Status](https://ci.ros2.org/job/ci_windows/11229/)

File: rosidl_typesupport_introspection_c/CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,10 @@ ament_index_register_resource("rosidl_runtime_packages")
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  find_package(rosidl_typesupport_interface REQUIRED)
+  # Give cppcheck hints about macro definitions coming from outside this package
+  get_target_property(
+    ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS rosidl_typesupport_interface::rosidl_typesupport_interface INTERFACE_INCLUDE_DIRECTORIES)
```
Comment (sloretz) [inline rosidl_typesupport_introspection_c/CMakeLists.txt:46]: note to self, `ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS` comes from here: https://github.com/ament/ament_lint/blob/8bf194aa1ac282db5483dd0d3fefff8f325b0db8/ament_cmake_cppcheck/cmake/ament_cppcheck.cmake#L18-L20

**Review by sloretz — APPROVED** [review]

**Comment by sloretz** [comment]
@claireyywang @nuclearsandwich does the CI branch https://github.com/ros2/ci/tree/revert-472-nuclearsandwich/pin-cppcheck need to be brought up to date before rerunning CI for this PR?

**Comment by nuclearsandwich** [comment]
> does the CI branch [https://github.com/ros2/ci/tree/revert-472-nuclearsandwich/pin-cppcheck](https://github.com/ros2/ci/tree/revert-472-nuclearsandwich/pin-cppcheck?rgh-link-date=2020-12-01T23%3A37%3A07Z) need to be brought up to date before rerunning CI for this PR?

Things have been pretty stable but I'll check and rebase if necessary then report back.

**Comment by clalancette** [comment]
We haven't moved on this in close to 2 years now, and in the meantime we've more-or-less deprecated cppcheck.  So I'm going to close this one out.

**Pull request closed without merging**